ddl/lightning/backend_mgr.go

```diff
@@ -0,0 +1,192 @@
+		if err != nil {
+			logutil.BgLogger().Warn(LitWarnConfigError, zap.Int64("job ID", jobID), zap.Error(err))
+			return nil, err
+		}
+		bd, err := createLocalBackend(ctx, cfg, glueLit{})
+		if err != nil {
+			logutil.BgLogger().Error(LitErrCreateBackendFail, zap.Int64("job ID", jobID), zap.Error(err))
+			return nil, err
+		}
+
+		bcCtx := newBackendContext(ctx, jobID, &bd, cfg, defaultImportantVariables, m.memRoot, m.diskRoot)
+		m.Store(jobID, bcCtx)
+
+		m.memRoot.Consume(StructSizeBackendCtx)
+		logutil.BgLogger().Info(LitInfoCreateBackend, zap.Int64("job ID", jobID),
+			zap.Int64("current memory usage", m.memRoot.CurrentUsage()),
+			zap.Int64("max memory quota", m.memRoot.MaxMemoryQuota()),
+			zap.Bool("is unique index", unique))
+		return bcCtx, nil
+	}
+	return bc, nil
+}
+
+func createLocalBackend(ctx context.Context, cfg *config.Config, glue glue.Glue) (backend.Backend, error) {
+	tls, err := cfg.ToTLS()
+	if err != nil {
+		logutil.BgLogger().Error(LitErrCreateBackendFail, zap.Error(err))
+		return backend.Backend{}, err
+	}
+
+	errorMgr := errormanager.New(nil, cfg, log.Logger{Logger: logutil.BgLogger()})
+	return local.NewLocalBackend(ctx, tls, cfg, glue, int(LitRLimit), errorMgr)
+}
+
+func newBackendContext(ctx context.Context, jobID int64, be *backend.Backend,
+	cfg *config.Config, vars map[string]string, memRoot MemRoot, diskRoot DiskRoot) *BackendContext {
+	bc := &BackendContext{
+		jobID:    jobID,
+		backend:  be,
+		ctx:      ctx,
+		cfg:      cfg,
+		sysVars:  vars,
+		diskRoot: diskRoot,
+	}
+	bc.EngMgr.init(memRoot, diskRoot)
+	return bc
+}
+
+// Unregister removes a backend context from the backend context manager.
+func (m *backendCtxManager) Unregister(jobID int64) {
+	bc, exist := m.Load(jobID)
+	if !exist {
+		return
+	}
+	bc.EngMgr.UnregisterAll(jobID)
+	bc.backend.Close()
+	m.memRoot.Release(StructSizeBackendCtx)
+	m.Delete(jobID)
+	m.memRoot.ReleaseWithTag(encodeBackendTag(jobID))
+	logutil.BgLogger().Info(LitInfoCloseBackend, zap.Int64("job ID", jobID),
+		zap.Int64("current memory usage", m.memRoot.CurrentUsage()),
+		zap.Int64("max memory quota", m.memRoot.MaxMemoryQuota()))
+}
+
+// TotalDiskUsage returns the total disk usage of all backends.
+func (m *backendCtxManager) TotalDiskUsage() uint64 {
+	var totalDiskUsed uint64
+	for _, key := range m.Keys() {
+		bc, exists := m.Load(key)
+		if exists {
+			_, _, bcDiskUsed, _ := bc.backend.CheckDiskQuota(math.MaxInt64)
+			totalDiskUsed += uint64(bcDiskUsed)
+		}
+	}
+	return totalDiskUsed
+}
+
+// UpdateMemoryUsage collects the memory usages from all the backend and updates it to the memRoot.
+func (m *backendCtxManager) UpdateMemoryUsage() {
+	for _, key := range m.Keys() {
+		bc, exists := m.Load(key)
+		if exists {
+			curSize := bc.backend.TotalMemoryConsume()
+			m.memRoot.ReleaseWithTag(encodeBackendTag(bc.jobID))
+			m.memRoot.ConsumeWithTag(encodeBackendTag(bc.jobID), curSize)
+		}
+	}
+}
+
+type glueLit struct{}
```

Comment:
Please add comments for `glueLit`. 
Besides, do we need to implement this struct's functions?

Reply:
> do we need to implement this struct's functions?

No, it is a noop struct.